cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -149,7 +149,14 @@ func ensureLayoutCompatibility(gadgetLayout *gadget.LaidOutVolume, diskLayout *p
 	eq := func(ds partition.DeviceStructure, gs gadget.LaidOutStructure) bool {
 		dv := ds.VolumeStructure
 		gv := gs.VolumeStructure
-		return dv.Name == gv.Name && ds.StartOffset == gs.StartOffset && dv.Size == gv.Size && dv.Filesystem == gv.Filesystem
+		check := dv.Name == gv.Name && ds.StartOffset == gs.StartOffset && dv.Filesystem == gv.Filesystem
+
+		// Don't test system-data size, it could have been expanded
+		if gv.Role == gadget.SystemData {
+			return check
```

Comment:
Should this be `check && dv.Size >= gv.Size` then?

Reply:
It makes sense to enforce the minimum size of the partition.